### DISCUSSION TITLE
Is the esp32 module compatible with a generic PS3 controller?

### DISCUSSION THREAD

**Discussion by BobEsponja12** [darthcloud/BlueRetro#476] [Q&A] [answered]
My question is if the esp32 module is compatible with the ps2 mode with a generic PS3 control

**Comment by vicsidious** [comment, marked as answer]
Yes it is! To pair a PS3 controller, you just have a few more steps to go through.
Here's a little tutorial to help you get going.
https://github.com/darthcloud/BlueRetro/wiki/Controller-pairing-guide#ps3
EDIT: Also, generic controllers are a bit "hit or miss". I paired 3 generic ps3 controllers successfully and one (on the cheaper side) failed miserably! Your mileage may vary!

**Reply by BobEsponja12** [reply]
Thanks a lot 
I hope it works with the control model that I have since I only had that doubt about the system

**Reply by vicsidious** [reply]
Keep in mind that an official ps3 controller isn't that expensive on Ebay, Craigslist or Marketplace, also DS4 are getting cheaper by the minute (even boxed DS5 were priced under  65$CAD last Friday!), so you shouldn't miss on BlueRetro because of that generic controller. Once you try it, you'll never be able to go back to wired controller. :)

**Comment by darthcloud** [comment]
Only official Sony PS3 controller are supported.

**Reply by vicsidious** [reply]
Without being supported, I observed that the somewhat well-made "Double Shock 3" with the "P3" home button do work with BlueRetro. Those are the 3 successful pair I spoke of earlier. I've used them on the PS2, but noticed some minor flaws: rumble doesn't seem to work and sometimes you have to fiddle with the "reset" button on the esp32 to finally sync the controller. Once connected, they do work as good as an official controller, and will let your little brother or your kid get his ass handed to him when playing DOA2 against you and your official controller XD.
Note that I understand that there might be multiple revisions and/or quality control (or lack thereof) with this particular generic model, and some may work and some not. Support extends only on official controllers probably for that very reason!
